R/select.R

```diff
@@ -0,0 +1,155 @@
+#' Select variables using `select`
+#'
+#' `step_select` creates a *specification* of a recipe step
+#'  that will select variables using [dplyr::select()].
+#'
+#' @inheritParams step_center
+#' @param ... One or more selector functions to choose which
+#'  variables that will be evaluated by the filtering bake. See
+#'  [selections()] for more details. For the `tidy`
+#'  method, these are not currently used.
+#' @param role For model terms selected by this step, what analysis
+#'  role should they be assigned?
+#' @param inputs Quosure(s) of `...`.
+#' @return An updated version of `recipe` with the new step
+#'  added to the sequence of existing steps (if any). For the
+#'  `tidy` method, a tibble with column `terms` which
+#'  contains the `select` expressions as character strings
+#'  (and are not reparsable).
+#' @details When an object in the user's global environment is
+#'  referenced in the expression defining the new variable(s),
+#'  it is a good idea to use quasiquotation (e.g. `!!`) to embed
+#'  the value of the object in the expression (to be portable
+#'  between sessions). See the examples.
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept variable_filters
+#' @export
+#' @examples
+#' rec <-
+#'   recipe(~., data = iris) %>%
+#'   step_select(Species, starts_with("Sepal"))
+#'
+#' prepped <- prep(rec, training = iris %>% slice(1:75))
+#'
+#' library(dplyr)
+#'
+#' dplyr_train <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(1:75) %>%
+#'   select(Species, starts_with("Sepal"))
+#'
+#' rec_train <- bake(prepped, new_data = NULL)
+#' all.equal(dplyr_train, rec_train)
+#'
+#' dplyr_test <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(76:150) %>%
+#'   select(Species, starts_with("Sepal"))
+#' rec_test <- bake(prepped, iris %>% slice(76:150))
+#' all.equal(dplyr_test, rec_test)
+#'
+#' # Embedding objects:
+#' sepal_vars <- c("Sepal.Width", "Sepal.Length")
+#'
+#' qq_rec <-
+#'   recipe(~., data = iris) %>%
+#'   # bad approach
+#'   step_select(Species, sepal_vars) %>%
+#'   # best approach
+#'   step_select(Species, !!sepal_vars) %>%
+#'   prep(training = iris)
+#'
+#' bake(qq_rec, new_data = NULL) %>% slice(1:4)
+#'
+#' # The difference:
+#' tidy(qq_rec, number = 1)
+#' tidy(qq_rec, number = 2)
+step_select <- function(recipe,
+                        ...,
+                        role = NA,
+                        trained = FALSE,
+                        inputs = NULL,
```

Comment:
I don't _think_ we need `inputs` as an argument here. I imagine that internally we can just reuse the `$terms` slot to hold the selected character vector of terms after calling `eval_select_recipes()`. The user would never use `inputs` here, and it would be overwritten even if they did

See below for some edits about how I'd do this

Reply:
:+1: I've changed it to use only `$terms` and removed `$inputs`